app/src/main/java/com/lucas/medicaltools/service/MedicalToolsInteractor.kt

```diff
@@ -0,0 +1,62 @@
+package com.lucas.medicaltools.service
+
+import android.util.Log
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import com.lucas.medicaltools.repository.MedicalToolsRepository
+import io.reactivex.Observable
+import io.reactivex.functions.Predicate
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import retrofit2.Response
+import java.lang.Exception
+
+object MedicalToolsInteractor {
+    private var medicalToolCache: List<MedicalTool>? = null
+    lateinit var storeToolCache: MutableList<MedicalTool>
+    lateinit var storeToolCacheCopy: MutableList<MedicalTool>
+
+    fun getMedicalTools(filter: String): Observable<MainViewState> {
+        return if (medicalToolCache == null ) {
+            MedicalToolsRepository().getMedicalTools()
```

Comment:
Your cache should manage all of the cache items, single object. If null, use it otherwise fire off the request.